File: python/euler/010.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-

def judge(n, lst):
	for i in lst:
		if n % i == 0:
			return False
	return True

def sum_prime(n):
	def prime(n):
		count = 2
		lst = [2]
		for i in xrange(3, n, 2):
			if judge(i, lst):
				lst.append(i)
				count += i
		return count
	return prime(n)

if __name__ == '__main__':
	print sum_prime(2000000)

File: python/euler/010.c
#include<stdio.h>
#define N 100000

long long int sum_prime(int n)
{
	long long int count = 2;
	int lst[N];
	int i, j;

	for(i = 0; i < n; i++){
		lst[i] = 1;
	}
	for(i = 3; i < n; i+=2){
		if(lst[i]){
			count += i;
			for(j = i*2; j < n; j+=i){
				lst[j] = 0;
			}
		}
	}
	return count;
}

int main(void)
{
	printf("%lld\n", sum_prime(2000000));

	return 0;
}
